pxtblocks/fields/field_tilemap.ts

```diff
@@ -329,4 +372,28 @@ namespace pxtblockly {
             }
         }
     }
+
+    function checkTilemap(tilemap: pxt.sprite.TilemapData, galleryItems: pxt.sprite.GalleryItem[]) {
+        if (!tilemap || !tilemap.tilemap || !tilemap.tilemap.width || !tilemap.tilemap.height) return false;
+
+        if (!tilemap.layers || tilemap.layers.width !== tilemap.tilemap.width || tilemap.layers.height !== tilemap.tilemap.height) return false;
+
+        if (!tilemap.tileset) return false;
+
+        for (const tile of tilemap.tileset.tiles) {
+            if (tile && (tile.projectId >= 0 || (tile.qualifiedName && galleryItems.some(g => g.qName === tile.qualifiedName)))) {
+                continue;
+            }
+
+            return false;
+        }
+
+        return true;
+    }
+
+    function checkTileset(tileset: pxt.sprite.TileSet, galleryItems: pxt.sprite.GalleryItem[]) {
```

Comment:
Is this here for future reference or something of the sort? Doesn't seem to be used.

Reply:
Whoops, this got refactored. I'll delete it